javascript/src/wombatSetup.js

```diff
@@ -0,0 +1,206 @@
+
+  // combine all decode parts to get the ZIM path
+  const zimPath = decoded_host + decoded_path + decoded_query;
+
+  // apply the fuzzy rules to the ZIM path
+  const fuzzifiedPath = applyFuzzyRules(zimPath);
+
+  // Reencode everything but '/' (we decode it afterwards for simplicity)
+  const finalUrl =
+    prefix + encodeURIComponent(fuzzifiedPath).replaceAll('%2F', '/');
+
+  /*
+  console.log(
+    'urlRewriten:\n\t- current_url: ' +
+      current_url +
+      '\n\t- orig_host: ' +
+      orig_host +
+      '\n\t- orig_scheme: ' +
+      orig_scheme +
+      '\n\t- orig_url: ' +
+      orig_url +
+      '\n\t- prefix: ' +
+      prefix +
+      '\n\t- url: ' +
+      url +
+      '\n\t- useRel: ' +
+      useRel +
+      '\n\t- mod: ' +
+      mod +
+      '\n\t- doc: ' +
+      doc +
+      '\n\t- finalUrl: ' +
+      finalUrl.toString() +
+      '\n\t',
+  );*/
+
+  return finalUrl;
+}
+
+export function getWombatInfo(
+  current_url, // The current (real) url we are on
+  orig_host, // The host of the original url
+  orig_scheme, // The scheme of the original url
+  orig_url, // The original url
+  prefix, // The (absolute) prefix to add to all our urls (from where we are served))
+) {
+  return {
+    // The rewrite function used to rewrite our urls.
+    rewrite_function: (url, useRel, mod, doc) =>
+      urlRewriteFunction(
+        current_url,
+        orig_host,
+        orig_scheme,
+        orig_url,
+        prefix,
+        url,
+        useRel,
+        mod,
+        doc,
+      ),
+
+    // Seems to be used only to send message to. We don't care ?
+    top_url: current_url,
+
+    // Seems to be used to generate url for blobUrl returned by SW.
+    // We don't care (?)
+    url: orig_url,
+
+    // Use to timestamp message send to top frame. Don't care
+    timestamp: '',
+
+    // Use to send message to top frame and in default rewrite url function. Don't care
+    request_ts: '',
+
+    // The url on which we are served.
+    prefix: prefix,
+
+    // The default mod to use.
+    mod: '',
+
+    // Use to detect if we are framed (and send message to top frame ?)
+    is_framed: false,
+
+    // ??
+    is_live: false,
+
+    // Never used ?
+    coll: '',
+
+    // Set wombat if is proxy mode (we are not)
+    proxy_magic: '',
+
+    // This is the prefix on which we have stored our static files (needed by wombat).
+    // Must not conflict with other url served.
+    // Will be used by wombat to not rewrite back the url
+    static_prefix: prefix + '_zim_static/',
+
+    wombat_ts: '',
+
+    // A delay is sec to apply to all js time (`Date.now()`, ...)
+    wombat_sec: 0,
+
+    // The scheme of the original url
+    wombat_scheme: orig_scheme,
+
+    // The host of the original url
+    wombat_host: orig_host,
+
+    // Extra options ?
+    wombat_opts: {},
+
+    // ?
+    enable_auto_fetch: true,
+    convert_post_to_get: true,
+    target_frame: '___wb_replay_top_frame',
+    isSW: true,
```

Comment:
What's this and why should it be true?

Reply:
I've opened https://github.com/openzim/warc2zim/issues/239